src/matrix/room/timeline/Timeline.js

```diff
@@ -216,6 +227,13 @@ export class Timeline {
         for (const entry of entries) {
             try {
                 this._remoteEntries.getAndUpdate(entry, Timeline._entryUpdater);
+                const oldEntry = this._contextEntriesNotInTimeline.get(entry.id)
```

Comment:
I wonder what will happen if you reply to a pending message (e.g. while offline), and then everything gets sent when  off-line. `id` returns the transaction id for `PendingEventEntry`s and the event_id for `EventEntry`s. Could it happen that we are left with both the PendingEventEntry and the EventEntry for the same message in the map, as they use a different id?

Reply:
Actually, in that scenario, the context entry will fail to load before it is sent, right? As:
 - it's not in remoteEntries
 - it's not in storage
 - it's not in the homeserver

Reply:
Can you test this scenario? We might need to also support getting the context entry from `this._localEntries` as well.

Reply:
Yes, the context entry would fail to load and this would render a `ReplyPreviewError`. When we're back online, the reply preview would be rendered. We need to extend ~~getTrackedEntry~~ findLoadedEventById to also look in localEntries for the context.

Reply:
> I wonder what will happen if you reply to a pending message (e.g. while offline), and then everything gets sent when off-line. `id` returns the transaction id for `PendingEventEntry`s and the event_id for `EventEntry`s. Could it happen that we are left with both the PendingEventEntry and the EventEntry for the same message in the map, as they use a different id?

As we discussed earlier, this should probably never happen since by definition a pending message would always be in timeline and therefore the map is never involved.